Index: pl.h
===================================================================
--- pl.h	(revision 3)
+++ pl.h	(working copy)
@@ -190,6 +190,7 @@
 #define clickmap_debug()				PL_clickmap_debug()
 #define clickmap_setdefaulturl( url )			PL_clickmap_setdefaulturl( url )
 #define clickmap_seturlt( url )				PL_clickmap_seturlt( url )
+#define clickmap_encodeurls ( mode )	    PL_clickmap_encodeurls ( mode )
 #define clickmap_entry( t, u, p, x1, y1, x2, y2, tp, cm, tit )  PL_clickmap_entry( t, u, p, x1, y1, x2, y2, tp, cm, tit )
 #define clickmap_out( tx, ty )				PL_clickmap_out( tx, ty )
 #define clickmap_show( dev )				PL_clickmap_show( dev )
@@ -262,6 +263,7 @@
 extern int PL_clickmap_debug();
 extern int PL_clickmap_setdefaulturl();
 extern int PL_clickmap_seturlt();
+extern int PL_clickmap_encodeurls(); /* added jqn 2/4/09 */
 extern int PL_clickmap_entry();
 extern int PL_clickmap_out();
 extern int PL_clickmap_show();
Index: clickmap.c
===================================================================
--- clickmap.c	(revision 3)
+++ clickmap.c	(working copy)
@@ -10,6 +10,10 @@
 
    Mar 27 2002 scg  	SVG clickable map support added (PLS.device == 's')  Not implemented here
 			but rather via SVG_beginobj and SVG_endobj.
+
+   Feb  4 2009 jqn      added encodeurl config parameter so that clickmap URL targets with embedded spaces
+                        will use '_' instead of '+'.
+
  */
 
 #include "pl.h"
@@ -32,6 +36,7 @@
 static char tpurl[MAXURL] = ""; /* a url template */
 static int mapstatus = 0; /* 1 if we are in the process of doing a map; 0 otherwise */
 static int demomode = 0;  /* 1 if we are in demo mode */
+static int encodeclickmapurls = 0; /* 1 if we want to use _ instead of + in clickmap URL targets - added jqn 2/4/09 */
 static int intersect = 0;
 static double adjx = 0.0, adjy = 0.0;
 
@@ -87,6 +92,7 @@
 for( i = 0; url[i] != '\0'; i++ ) {
 	if( url[i] == ' ' || url[i] == '\n' ) url[i] = '_';
 	if( url[i] == '\\' && url[i+1] == 'n' ) { url[i++] = '_'; url[i] = '_'; }
+	if( encodeclickmapurls && url[i] == '+' ) url[i] = '_';  /* added jqn 2/4/09 */
 	}
 
 urls[ imap ] = (char *) malloc( strlen( url ) + 1 );
@@ -444,6 +450,16 @@
 return( 0 );
 }
 
+/* ========================== */
+/* ENCODEURLS - make sure the clickmap URLs use _ instead of + to represent embedded spaces - added jqn 2/4/09  */
+int
+PL_clickmap_encodeurls ( mode )
+int mode;
+{
+encodeclickmapurls = mode;
+return( 0 );
+}
+
 /* =========================== */
 static int
 get_targetstr( buf, targetstr )
Index: proc_settings.c
===================================================================
--- proc_settings.c	(revision 3)
+++ proc_settings.c	(working copy)
@@ -82,6 +82,11 @@
         else PL_encode_fnames( 0 );
 	}
 
+else if( strcmp( attr, "encodeclickmapurls" )==0 ) {					      /* added jqn 2/4/09 */
+        if( strncmp( val, YESANS, 1 )==0 ) PL_clickmap_encodeurls( 1 ); 
+        else PL_clickmap_encodeurls( 0 );
+	}
+
 #ifndef NOPS
 else if( strcmp( attr, "ps_latin1_encoding" )==0 ) {
         if( strncmp( val, YESANS, 1 )==0 ) PLGP_settings( "ps_latin1_encoding", "1" ); /* added 7/28/04 */
